test/helpers/cilium.go

```diff
@@ -343,14 +359,74 @@ func (c *Cilium) ServiceAdd(id int, frontend string, backends []string, rev int)
 	return c.Exec(cmd)
 }
 
+// ServiceDel deletes the service with the specified ID
+func (c *Cilium) ServiceDel(id int) *CmdRes {
+	return c.Exec(fmt.Sprintf("service delete '%d'", id))
+}
+
+// ServiceIsSync checks that the service is sync in the bpf lb list
+func (c *Cilium) ServiceIsSync(id int) (bool, error) {
+	var svc *models.Service
+	svcRes := c.ServiceGet(id)
+	if !svcRes.WasSuccessful() {
+		return false, fmt.Errorf("cannot get service id %d: %s", id, svcRes.CombineOutput())
+	}
+	err := svcRes.UnMarshal(&svc)
+	if err != nil {
+		return false, err
+	}
+
+	bpfLB, err := c.BpfLBList()
+	if err != nil {
+		return false, err
+	}
+
+	frontendAddr := net.JoinHostPort(
+		svc.FrontendAddress.IP,
+		fmt.Sprintf("%d", svc.FrontendAddress.Port))
+	lb, ok := bpfLB[frontendAddr]
+	if ok == false {
+		return false, fmt.Errorf("service is not sync with BPF lb")
+	}
+
+	for _, backendAddr := range svc.BackendAddresses {
+		result := false
+		backendSVC := net.JoinHostPort(
+			*backendAddr.IP,
+			fmt.Sprintf("%d", backendAddr.Port))
+		target := fmt.Sprintf("%s (%d)", backendSVC, id)
+
+		for _, addr := range lb {
+			if addr == target {
+				result = true
+			}
+		}
+		if result == false {
+			return false, fmt.Errorf("backend addr %s does not match with LB", target)
```

Comment:
addr --> address

Reply:
"does not match" is unclear. specify that the backend address does not exist in the BPF maps for the specified frontend and provide context for which service this has occurred.